src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpCompilerOptionsTests.cs

```diff
@@ -167,5 +170,59 @@ public void ProjectCompilationOutputsChange()
             outputs = (CompilationOutputFilesWithImplicitPdbPath)environment.Workspace.GetCompilationOutputs(project.Test_VisualStudioProject.Id);
             Assert.Equal(@"C:\a\3.dll", outputs.AssemblyFilePath);
         }
+
+        [WpfTheory]
+        [InlineData(LanguageVersion.CSharp7_3)]
+        [InlineData(LanguageVersion.CSharp8)]
+        [InlineData(LanguageVersion.Latest)]
+        [InlineData(LanguageVersion.LatestMajor)]
+        [InlineData(LanguageVersion.Preview)]
+        [InlineData(null)]
+        public void SetProperty_MaxSupportedLangVersion(LanguageVersion? maxSupportedLangVersion)
+        {
+            const LanguageVersion attemptedVersion = LanguageVersion.CSharp8;
+
+            var catalog = TestEnvironment.s_exportCatalog.Value
+                .WithParts(
+                    typeof(CSharpParseOptionsChangingService));
+
+            var factory = ExportProviderCache.GetOrCreateExportProviderFactory(catalog);
+            using var environment = new TestEnvironment(exportProviderFactory: factory);
+
+            var hierarchy = environment.CreateHierarchy("CSharpProject", "Bin", projectRefPath: null, projectCapabilities: "CSharp");
+
+            if (!(hierarchy is IVsBuildPropertyStorage storage))
```

Comment:
Either do a hard cast or Assert.IsAssignableFrom -- that way the diagnostic if this isn't implemented will point to the problem better than "expected true, got false". The IsAssignableFrom also returns the object in it's casted form.